File: src/main/java/io/github/marcuscastelo/invtweaks/inventory/ScreenInventory.java
package io.github.marcuscastelo.invtweaks.inventory;

import net.minecraft.screen.ScreenHandler;

public class ScreenInventory {
    public final ScreenHandler screenHandler;
    public final int start, end;

    public ScreenInventory(ScreenHandler screenHandler, int start, int end) {
        this.screenHandler = screenHandler;
        this.start = start;
        this.end = end;
    }

    public int getSize() {
        return end-start+1;
    }
}

File: src/main/kotlin/io/github/marcuscastelo/invtweaks/inventory/ScreenInventory.kt
package io.github.marcuscastelo.invtweaks.inventory

import net.minecraft.screen.ScreenHandler

data class ScreenInventory(val screenHandler: ScreenHandler, val start: Int, val end: Int) {
    val size = end - start + 1

    val slotRange = start..end
    val slots get() = screenHandler.slots.subList(start, end + 1)
    val stacks get() = screenHandler.stacks.subList(start, end + 1)
}
